src/test/groovy/org/prebid/server/functional/tests/privacy/AuctionGppSpec.groovy

```diff
@@ -0,0 +1,226 @@
+        given: "Default bid request with gpp and gppSid"
+        def gppConsent = new TcfEuV2Consent.Builder().build()
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gppSid = [TCF_EU_V2.valueAsInt]
+            regs.gpp = gppConsent
+            user = new User().tap {
+                consent = null
+            }
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain user.consent from regs.gpp"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gpp == gppConsent.consentString
+        assert bidderRequest.user.consent == gppConsent.toString().substring(7)
+    }
+
+    def "PBS should emit warning when gppSid contains 2, gpp is TCF2-EU and regs.gpp and user.consent are different"() {
+        given: "Default bid request with gpp and gppSid"
+        def validConsentString = new TcfConsent.Builder()
+                .setPurposesLITransparency(BASIC_ADS)
+                .addVendorLegitimateInterest([GENERIC_VENDOR_ID])
+                .build()
+        def gppSidsId = [TCF_EU_V2.valueAsInt]
+        def gpp = new TcfEuV2Consent.Builder().build()
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gpp: gpp, gppSid: gppSidsId)
+            user = new User(consent: validConsentString)
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP TCF2 string does not match user.consent"]
+
+        and: "Bidder request should contain the same value as in request"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gpp == gpp as String
+        assert bidderRequest.regs.gppSid == gppSidsId
+        assert bidderRequest.user.consent == validConsentString as String
+    }
+
+    def "PBS should copy regs.gpp to regs.usPrivacy when gppSid contains 6, gpp is USP_V1 and regs.us_privacy isn't specified"() {
+        given: "Default bid request with gpp and gppSid, without us_privacy"
+        def gppConsent = new UspV1Consent.Builder().build()
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gpp: gppConsent, gppSid: [USP_V1.valueAsInt], usPrivacy: null)
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.usPrivacy from regs.gpp"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.usPrivacy == gppConsent.toString().substring(7)
```

Comment:
What is `.substring(7)`? This is pure magic. Add a method to our GPP wrapper to encode only the consent part of the GPP.